RockPaperScissors/RockPaperScissors/main.swift

```diff
@@ -1,10 +1,96 @@
 //
 //  RockPaperScissors - main.swift
-//  Created by yagom. 
+//  Created by Lingo, 우롱차.
 //  Copyright © yagom academy. All rights reserved.
 // 
 
 import Foundation
 
-print("Hello, World!")
+enum Settings {
+    static let exitCode = 0
+}
 
+enum GameDisplayMessage {
+    static let menu = "가위(1), 바위(2), 보(3)! <종료 : 0> : "
+    static let gameDidEnd = "게임 종료"
+    static let playerDidWin = "이겼습니다!"
+    static let playerDidLose = "졌습니다!"
+    static let playerDidDraw = "비겼습니다!"
+    static let invalidPlayerInput = "잘못된 입력입니다. 다시 시도해주세요."
+    static let error = "에러가 발생했습니다. 게임을 다시 실행시켜 주세요."
+}
+
+enum HandType: Int, CaseIterable {
+    case scissor = 1
+    case rock = 2
+    case paper = 3
+    
+    static var randomHandType: Int? {
+        return Self.allCases.randomElement()?.rawValue
+    }
+}
+
+enum GameResult {
+    case playerWin
+    case playerLose
+    case playerDraw
+}
+
+func getPlayerInput() -> Int? {
+    guard let playerInput = readLine(),
+          let playerIntInput = Int(playerInput),
+          let playerGameOptionsInput = HandType(rawValue: playerIntInput),
+          (HandType.allCases.contains(playerGameOptionsInput) || playerIntInput == Settings.exitCode) else {
+        return nil
+    }
```

Comment:
변환할 때 마다 새로운 `상수`로 이름지어 받아주는 점, 꼼꼼하고 좋습니다.

다만, 하나의 guard문 안에 너무 많은 일들이 모여있으면, 오히려 가독성이 떨어질 수 있습니다.
특히 `옵셔널 바인딩`을 하는 guard문과, 조건문 역할을 하는 guard문은 분리해주는 게 좋습니다. 😄 

```suggestion
    guard let playerInput = readLine(),
          let playerIntInput = Int(playerInput),
          let playerGameOptionsInput = HandType(rawValue: playerIntInput) else {
        return nil
    }
    
    guard HandType.allCases.contains(playerGameOptionsInput) || playerIntInput == Settings.exitCode else {
        return nil
    }
```

Reply:
guard 문 분리 완료했습니다! 😃